backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,275 @@
+package backtraceio.library.breadcrumbs;
+
+import android.content.Context;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import java.io.File;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import backtraceio.library.models.types.BacktraceBreadcrumbLevel;
+import backtraceio.library.models.types.BacktraceBreadcrumbType;
+
+public class BacktraceBreadcrumbs {
+
+    /**
+     * Which breadcrumb types are enabled?
+     */
+    private Set<BacktraceBreadcrumbType> enabledBreadcrumbTypes;
+
+    /**
+     * The Backtrace BroadcastReciever instance
+     */
+    private BacktraceBroadcastReceiver backtraceBroadcastReceiver;
+
+    /**
+     * The Backtrace ComponentCallbacks2 listener
+     */
+    private BacktraceComponentListener backtraceComponentListener;
+
+    /**
+     * The base directory of the breadcrumb logs
+     */
+    private String breadcrumbLogDirectory;
+
+    public BacktraceBreadcrumbs(Context context) {
+        // Create the breadcrumbs subdirectory for storing the breadcrumb logs
+        breadcrumbLogDirectory = context.getFilesDir().getAbsolutePath() + "/breadcrumbs";
+        File breadcrumbLogsDir = new File(breadcrumbLogDirectory);
+        breadcrumbLogsDir.mkdir();
+    }
+
+    private void enableSystemBreadcrumbs(Context context) {
+        backtraceBroadcastReceiver = new BacktraceBroadcastReceiver(this);
+        context.registerReceiver(backtraceBroadcastReceiver, backtraceBroadcastReceiver.getMyIntentFilter());
+
+        backtraceComponentListener = new BacktraceComponentListener(this);
+        context.registerComponentCallbacks(backtraceComponentListener);
+    }
+
+    private void enableBreadcrumbType(Context context, BacktraceBreadcrumbType type) {
+        switch (type)
```

Comment:
Currently only the `SYSTEM` breadcrumbs require a special integration which needs to be enabled/disabled in a special way, because of `BroadcastReceiver` and `ComponentCallbacks2`. This may change in the future as we add more Android listeners/integrations or refactor our current integrations.

Reply:
Outdated, hopefully this is more clear now